MapboxCoreNavigation/PassiveLocationDataSource.swift

```diff
@@ -117,7 +117,8 @@ extension PassiveLocationDataSource: CLLocationManagerDelegate {
             return
         }
         
-        let status = navigator.status(at: lastRawLocation.timestamp)
+        let timestampForStatus = Date().addingTimeInterval(RouteControllerDeadReckoningTimeInterval)
+        let status = navigator.status(at: timestampForStatus)
```

Comment:
PassiveLocationManager is map view–specific, but PassiveLocationDataSource is not. To decouple map view behavior from location tracking behavior, we’d need PassiveLocationDataSource to expose a method similar to `RouteController.projectedLocation(for:)`, with all the same caveats.